pkg/webui/src/JobList.tsx

```diff
@@ -68,9 +70,13 @@ class JobListImpl extends React.Component<JobListProps, JobListState> {
     protected startListening() {
         try {
             const req = new SubscribeRequest();
-            let evts = this.props.client.subscribe(req);
-            evts.on('end', () => setTimeout(() => this.startListening(), 1000));
-            evts.on('data', r => {
+            req.setFilterList(this.state.search);
+            if (this.eventStream) {
+                this.eventStream.cancel();
+            }
```

Comment:
nit: I'd cancel the stream first and then create the request (i.e. move line 73 behind the if). This way request creation and use are closer together.